ddl/partition.go

```diff
@@ -3313,6 +3328,76 @@ func checkExchangePartitionRecordValidation(w *worker, pt *model.TableInfo, inde
 	return nil
 }
 
+func verifyExchangePartitionRecordCheckConstraint(w *worker, pt, nt *model.TableInfo, pschemaName, nschemaName, partitionName string) error {
+	getWriteableConstraintExpr := func(constraints []*model.ConstraintInfo) []string {
+		writeableConstraintExpr := make([]string, 0, len(constraints))
+		for _, con := range constraints {
+			if !con.Enforced {
+				continue
+			}
+			if con.State == model.StateDeleteOnly || con.State == model.StateDeleteReorganization {
+				continue
+			}
+			writeableConstraintExpr = append(writeableConstraintExpr, con.ExprString)
+		}
+		return writeableConstraintExpr
+	}
```

Comment:
Would it be possible to reuse `TableCommon.WritableConstraints()` instead?

Reply:
If add a interface for "TableCommon.WritableConstraints" it will ok. 
I did not find a way to directly change table to TableCommon and then call WritableConstraints.
I add a interface wrapper in the other pr.